lib/src/hooks/useDynamicForm.tsx

```diff
@@ -175,59 +182,23 @@ export function useDynamicForm(sections: DynamicFormSection[]): UseDynamicForm {
         return initialFormState;
     }, []);
 
-    const [state, setState] = useState<DynamicFormInternalState>(initialState);
-    const [errorMessages, setErrorMessages] = useState<DynamicFormErrors>({});
-    const [warningMessages, setWarningMessages] = useState<DynamicFormWarnings>({});
-    const [dirty, setDirty] = useState<boolean>(false);
-
-    /**
-     * SHOULD RENDER CONTROL:
-     * Determines if the form control should be rendered or not.
-     * @param sectionKey section key
-     * @param itemKey section key
-     */
-    const shouldRender: ShouldRenderFormItem = useCallback<ShouldRenderFormItem>(
-        (sectionKey: string, itemKey: string): boolean => {
-            const { rulerKey, condition, controlType }: Partial<DynamicFormItem> =
-                sections?.find((item: DynamicFormSection) => item.key === sectionKey)?.items?.find((item: DynamicFormItem) => item.key === itemKey) || {};
-            if (controlType === "Hidden") {
-                // Marked as hidden, don't render
-                return false;
-            }
-            if (typeof rulerKey === "string" && !!rulerKey.length) {
-                let rulerState: DynamicFormInternalStateValue;
-
-                if (!isEmpty(state) && !isEmpty(state[sectionKey])) {
-                    rulerState = (state[sectionKey] as DynamicFormInternalStateSection)[rulerKey];
-                }
+    const initialVisibility: DynamicFormVisibilityState = useMemo(() => {
+        const initialVisibilityState: DynamicFormVisibilityState = {};
+        sections?.map((section: DynamicFormSection) => {
+            initialVisibilityState[section?.key] = {};
+            section.items?.map((item) => {
+                const { initiallyHidden, key }: DynamicFormItem = item;
+                initialVisibilityState[section?.key][key] = !initiallyHidden;
```

Comment:
optional, we could remove the `optional` operator on line #188 and #191 as line #187 pretty much guard against undefined `section`